### DISCUSSION TITLE
[Feature] Preview generation for RAW photos edited in Darktable

### DISCUSSION THREAD

**Discussion by frigi83** [immich-app/immich#8848] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

To begin with, I would like to express my sincere appreciation for creating such a remarkable project focused on organizing, accessing, and protecting valuable memories.

Here is a suggestion on how Immich could be improved to better serve users who prefer editing their RAW photos in Darktable: At present, when I add pictures to Immich after making adjustments in Darktable, it utilizes the original RAW format to generate previews. Although this approach is logical from a technical standpoint, there are instances where it would be beneficial if the previews reflected the final edits rather than merely displaying the out-of-the-camera shots.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
#5758, #8773

**Reply by frigi83** [reply]
> #5758, #8773

The existing requests utilize the RAW file previews and not the edits made with another software which does not modify the actual RAW file but rather uses a sidecar containing the modifications.

We request the reopening of this feature request.

**Reply by alextran1502** [reply]
@frigi83 Do you mean using the sidecar to applied the edit information to the thumbnail or the RAW file?

**Reply by frigi83** [reply]
> @frigi83 Do you mean using the sidecar to applied the edit information to the thumbnail or the RAW file?

Yes and display in immich the "darktable image" instead of the original RAW

If you want I can provide some files as example in the next days.

**Comment by develar** [comment]
I believe this feature request is not about using an embedded preview, but rather about using [darktable-cli](https://docs.darktable.org/usermanual/4.0/en/special-topics/program-invocation/darktable-cli/) to render a RAW file with an XMP file. It's the only method that can correctly render an edited RAW file (Darktable also has some level of Adobe XMP (e.g. Lightroom) support).

As photoprism does (according to docs) — https://docs.photoprism.app/developer-guide/metadata/xmp/#raw-conversion

**Reply by frigi83** [reply]
Just wanted to clarify what I was saying in my previous message. Basically, you can use Darktable to edit your pics - crop 'em, adjust colors, etc., no problem. But here's the deal, those edits won't show up on Immich because Darktable doesn't actually change the RAW file itself. 

How it is possible to reopen this feature?

**Comment by jeena** [comment]
I agree that this feature request is very different to the linked ones. It's about using the sidecar file to crop and apply other adjustments used to edit it. I'm doing it in my other script to create "thumbnails" https://github.com/jeena/synology-pictures/blob/main/darktable.py#L16

The only other way achieving the same now is to not pull in the RAW files at all and instead to export each and every picture and only use those in immich.

I'm also voting for reoppening this issue.

**Comment by fpottbaecker** [comment]
I would second previous points on this being different from extracting the RAW preview, since DarkTable does not touch the RAW files.

Although, this could perhaps be formulated differently, not just for Darktable, but a new concept in the preview generation pipeline:  
Instead of just generating thumbnails from images, supported RAW files / negatives (with supported sidecards) can be developed into positives as a cachable step, and further jobs, like thumbnail generation then uses the positives instead of the negatives.
This could also (in a further down the road feature) result in a single negative creating multiple positives, c.f. DarkTable duplicate function where both sidecars derive from the same negative [c.f. xmpMM:derivedFrom](https://developer.adobe.com/xmp/docs/XMPNamespaces/xmpMM/).

**Comment by WrobotGames** [comment]
Hi, can this discussion be reopened? This is still a feature multiple people are interested in, and it is not a duplicate of anything else. 

For me, this feature would be useful if I want to have up to date versions of my edits on immich without having to reexport and replace the previous jpeg manually every single time. Another thing is that I do not need to have jpeg versions of the photos on my hard drive. 

As a sidenote, in photoprism, you get to choose between Darktable and RawTherapee as a converter.

**Reply by fpottbaecker** [reply]
I would definitely second this notion.
I also understand, that doing it nicely is probably a rather complicated technical undertaking. But still a discussion worth having.

**Reply by shrippen** [reply]
+1

**Reply by frigi83** [reply]
I thought about re-submitting the discussion with some examples, but the situation at home delayed everything.

With version [2.3.0 Immich](https://github.com/immich-app/immich/releases/tag/v2.3.0) introduced (under the hood) workflows, and I was wondering if that could be an alternative solution.

**Reply by fpottbaecker** [reply]
Oh, that looks like an option, if the system can handle this nicely (e.g. consider an XMP change an Asset upload).
Maybe this could warrant opening something else, I don't think this one is gonna get reopened.

**Comment by sleekspeed22** [comment]
Since Immich is the gallery of you photos and video.  I would think when sharing or viewing you'd want to show the corrected (color, cropping, etc) versions.  Building thumbnails based on the edited versions seems like the best way to show/see your work/photos in the best light.  Maybe to satisfy all... this can be a setting for thumbnails... create based on RAW or include sidecar edits. And/Or a toggle anywhere you are viewing groups of photos/videos.  I know I'll always choose side care edits... then the download should let you download image with the sidecar. Especially for but not limited to slideshows. 

I know it's not a small ask but I suspect it will be immensely useful to those of use that batch edit photo in darktable but use Immich for cataloguing, viewing, searching, and sharing.
